src/Files.App/UserControls/TabBar/TabBar.xaml.cs

```diff
@@ -84,6 +89,40 @@ public TabBar()
 			};
 		}
 
+		/// <summary>
+		/// Initializes the title bar asynchronously. This method attempts to get the title bar's left and right insets.
+		/// If the values are properly initialized, it sets the <see cref="TitleBarWidth"/> property. If the initialization fails,
+		/// it falls back to a default width.
+		/// </summary>
+		private async void InitializeTitleBarAsync()
```

Comment:
Do we still need multiple attempts now that you added `UIElement.Measure`?

Reply:
UIElement.Measure I added this before this test. But then you described a problem that never appeared during my testing. If this problem was only caused by a missing UIElement.Measure, then this test would no longer be necessary.

Reply:
Let's see what happens if we remove it, hopefully `UIElement.Measure` is all we need.

Reply:
In my personal testing, this has not manifested itself, but I plan to test this further on a weaker laptop where slower loading and thus the problem you describe may manifest itself.